routes/project/discount_code.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+class Clover
+  hash_branch(:project_prefix, "discount-code") do |r|
+    r.web do
+      authorize("Project:billing", @project.id)
+      billing_path = "#{@project.path}/billing"
+
+      r.post true do
+        discount_code = r.params["discount_code"].to_s.strip.downcase
+        Validation.validate_short_text(discount_code, "discount_code")
+
+        # Check if the discount code exists
+        discount = DiscountCode.where(Sequel.function(:lower, :discount_code) => discount_code).first
```

Comment:
If you change the `discount_code` column type to `citext`, you won't need to use the lower function.
Example usage: https://github.com/ubicloud/ubicloud/commit/c427e67871cfdc12502780d2263bfc414aec43bb

```suggestion
        discount = DiscountCode.first(discount_code: discount_code)
```

Reply:
Done.